### PR TITLE
Adds New `XrpError` Type Specific to Protocol Buffer Conversions

### PR DESCRIPTION
**Pull request by amiecorso**
## High Level Overview of Change
This PR adds a new type to the `XrpError` class in preparation for throwing descriptive errors from the Protobuf wrapper classes.  Note that while a pre-packaged `XrpError.malformedProtobuf` object exists as a static attribute with default error message on `XrpError`, we encourage the use of custom instances of `XrpError` with type `XrpTransactionType.MalformedProtobuf` and a more descriptive error message. 

### Context of Change
We are about to audit our Protobuf wrapper classes in `xpring-js` for more robust logic checks, more useful throwing behavior, and more descriptive errors.

### Type of Change

- [x] New feature (non-breaking change which adds functionality)

## Before / After
`XrpErrorType` has a new enum member, `XrpError` has a new static attribute.

## Test Plan
CI passes

### CODE REVIEW

**Review by mvadari — APPROVED** [review]

File: src/XRP/xrp-error.ts [hunk 1]
```diff
@@ -30,6 +31,14 @@ export default class XrpError extends Error {
     'Could not convert payment transaction: (transaction). Please file a bug at https://github.com/xpring-eng/Xpring-JS/issues',
   )
 
+  /**
+   * Encountered a protocol buffer formatted in contradiction to the logic of the XRPL.
```
Comment (tedkalaw) [inline src/XRP/xrp-error.ts:35]: can we link to the docs?

File: src/XRP/xrp-error.ts [hunk 2]
```diff
@@ -30,6 +31,14 @@ export default class XrpError extends Error {
...
+   */
+  static malformedProtobuf = new XrpError(
+    XrpErrorType.MalformedProtobuf,
+    'Encountered a protocol buffer formatted in contradiction to the XRPL business logic',
```
Comment (tedkalaw) [inline src/XRP/xrp-error.ts:39]: perhaps "Encountered a protocol buffer with unexpected format" or something similar? i feel this error message could be simpler

**Review by tedkalaw — APPROVED** [review]
seems reasonable, one comment inline

File: src/XRP/xrp-error.ts (hunk 1 above)
Comment (amiecorso) [reply src/XRP/xrp-error.ts:35]: good idea.  these'll have to be the generic xrpl.org docs

**Review by mvadari — APPROVED** [review]

**Pull request merged**